fimdp/core.py

```diff
@@ -1005,3 +1006,50 @@ def _next_action(self):
         actions = self.mdp.actions
         a_id = self.mdp.actions_for_state(self._current_state).next
         return actions[a_id]
+
+class Simulator():
+    """Class for simulating a strategy object on a ConsMDP.
+    
+    Picks actions based on given strategy for `duration` number of simulation
+    steps and stores the state and action history for further analysis. Interface
+    allows for extending simulation and resetting given instance using `simulate`
+    and `reset` methods.
+    """
+    
+    def __init__(self, strategy, duration):
+        
+        self.state_history = [strategy._current_state]
+        self.action_history = []
+        self.strategy = strategy
+        self.simulate(duration)
+        
+    def simulate(self, duration):
+        for step in range(duration):
+            actionobj = self.strategy.next_action()
+            next_state = self._next_state(actionobj)
+            self.strategy.update_state(next_state)
+            self.state_history.append(next_state)
+            self.action_history.append(actionobj.label)
+                
+    def reset(self, strategy=None):
```

Comment:
I have changed the interface to reset the current strategy instead of supplying a new one. That seems more natural to me.